### PR TITLE
Implement UART driver files for AT32UC3B: Working commit

### PR DESCRIPTION
**Pull request by ramangopalan**
## 拉取/合并请求描述：(PR description)

[

Dear RT-Thread Community, greetings! Firstly, many thanks for
RT-Thread! I am currently attempting to get RT-Thread onto the Mizar32
platform [1] which hosts the AT32UC3A series of microcontrollers.

为什么提交这份PR (why to submit this PR)

But before, I wanted to evaluate the current state of the AVR32
codebase. This commit cleans up the build for AT32UC3B0. I plan to use
this as a base for the work on AT32UC3A0*.

你的解决方案是什么 (what is your solution)

I checked many of the directories within bsp/ to compare against the
current (rather old) AVR32 implementation. This commit fixes
compilation problems for the at32uc3b0 target by implementing a UART
driver (among other things).

在什么测试环境下测试通过 (what is the test environment)

I wrote the drivers on a Windows machine - because of lack of debug
capability on my GNU/Linux platform. I used the standard AVR Atmel ICE
programmer for programming and debugging. I have a board from MCUZone
(donated to me by the SimpleMachines community - thank you Sergio :) )
which hosts the AVR32UC3B0256 microcontroller.

I checked the PR with this target. It works well. Unfortunately, the
links for the development board on MCUZone OR Aliexpress don't work
anymore. Luckily, I found a Youtube video [2]. This board/target,
exactly.

Finally, please consider merging this PR. I wish to frequent the
RT-Thread codebase for more hacks and adventures in the context of
AVR32. Please let me know if there's something I have missed OR if I
could have something differently.

Many thanks again for RT-Thread!

    R

References:
[1]: https://simplemachines.it/index.php/historical-projects/mizar-project/111-simplemachines/projects/mizar32-projects
[2]: https://www.youtube.com/watch?v=iv0rM-VFsKw

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [x] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 代码是高质量的 Code in this PR is of high quality
- [x] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

**Comment by supperthomas** [comment]
Thank you for contribute. Can you add readme.md for other to setup the environment of AVR32 easily? like 
https://github.com/RT-Thread/rt-thread/blob/master/bsp/core-v-mcu/core-v-cv32e40p/readme_EN.md

**Review by supperthomas — APPROVED** [review]

File: bsp/avr32uc3b0/application.c [hunk 1]
```diff
@@ -37,12 +37,12 @@ void rt_thread_entry_led2(void* parameter)
     while (1)
     {
         /* led on */
-        rt_kprintf("LED2 on");
+        //rt_kprintf("LED2 on");
```
Comment (supperthomas) [inline bsp/avr32uc3b0/application.c:40]: remove it or release it. Please don't "//" it .

**Review by supperthomas — APPROVED** [review]

File: bsp/avr32uc3b0/application.c (hunk 1 above)
Comment (ramangopalan) [reply bsp/avr32uc3b0/application.c]: Dear @supperthomas, I have removed the commented rt_kprintf statements. Thank you.

**Review by mysterywolf — APPROVED** [review]

**Comment by mysterywolf** [comment]
Thanks for you PR!

**Comment by ramangopalan** [comment]
Thank you @mysterywolf. Good day!

**Pull request merged**